src/libaktualizr/package_manager/ostreemanager.cc

```diff
@@ -395,7 +395,13 @@ Uptane::Target OstreeManager::getCurrent() const {
   // themselves, this actually works just fine for them, too.
   storage_->loadPrimaryInstalledVersions(&current_version, nullptr);
 
-  if (!!current_version && current_version->sha256Hash() == current_hash) {
+  bool staged_reboot_pending = false;
```

Comment:
Nice, thanks for this fix.

I think we should move all required functionality from the aktualizr's ostreemanager into the aklite's rootfsmanager going further in order to get rid of dependency on the package manager interface at all as well as on aktualizr in general. Not related to this PR, just strategic direction.

Reply:
This change seems to be break aklite unit tests:
```
The following tests FAILED:
        114 - test_aklite_rollback (Failed)
        115 - test_aklite_rollback_ext (Failed)
        119 - test_aklite_offline (Failed)
        121 - test_cli (Failed)
```
I need to take a better look to understand the reason.

Reply:
Because those tests are adjusted for "getCurrent" returning invalid current target in non-booted environment.

We need either:

a) adjust these tests to this change;
b) revert this change, and adjust the e2e test somehow so we can get desired behavior in the e2e test environment. Maybe we can play with `ostree admin deploy` command or something like that to make ostreemanager change its opinion about current target :).

Reply:
I've dropped the change from this PR for now, so we can merge the user rollback feature regardless of it.